### PR TITLE
docs: add 0031 new monitor histogram

### PR DESCRIPTION
**Pull request by zmstone**

### CODE REVIEW

File: active/0031-new-monitor-histogram.md [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+## Abstract
+
+This EIP proposes to improve the monitor histogram implementation to make it more compact and efficient.
+
+## Motivation
+
+The current (as of 5.8) monitor histogram has below issues:
+
+- It stores too much data but in the end only needed 1000 data points to return.
+- The talbe is of type `set` making it impossible to perform in-place compaction.
```
Comment (ieQu1) [inline active/0031-new-monitor-histogram.md:16]:
```suggestion
- The table is of type `set` making it impossible to perform in-place compaction.
```

File: active/0031-new-monitor-histogram.md [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+- The data points are formated as JSON array with repeated field names.
+
+All these issues make the current implementation inefficient and slow and reflected as timeouts in the dashboard.
+
+## Design
+
+- Create a new `ordered_set` table, key being `{Timestamp, node()}` and value being the counters and gauges in a map.
+- Do not use `local_content` for the table, so there is no need to RPC when querying the data.
+- For counters, store absolute values instead of calculating deltas. This makes compaction easier: simply downsample the data points.
+- Downsample the data points to less than 1000 points in the table. See downsampling algorithm below.
```
Comment (ieQu1) [inline active/0031-new-monitor-histogram.md:27]: Which node is responsible for downsampling the data? Each node for its own?

File: active/0031-new-monitor-histogram.md [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+- For the last 1 week, keep 1 data point per 1 hour, (6 * 24 = 144) data points in total.
+
+The total number of downsampling data points should not exceed 700.
+
+When downsampling, the data point which is earlier but closest to a wall clock aligned interval should be kept.
+
+For example:
+- `12:35:00` when interval is 10s.
+- `12:40:00` when interval is 10m.
+- `13:00:00` when interval is 1h.
```
Comment (thalesmg) [inline active/0031-new-monitor-histogram.md:45]: I don't understand this example.  Maybe provide more context and the hypothetical current time?

**Review by thalesmg — DISMISSED** [review]

File: active/0031-new-monitor-histogram.md (hunk 2 above)
Comment (zmstone) [reply active/0031-new-monitor-histogram.md:27]: each node itself.

File: active/0031-new-monitor-histogram.md (hunk 3 above)
Comment (zmstone) [reply active/0031-new-monitor-histogram.md]: updated.

File: active/0031-new-monitor-histogram.md [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+
+### Downsampling Algorithm
+
+The downsampling algorithm is as follows:
+
+- For the last 1 hour, keep 1 data point per 10 seconds, (60 * 6 = 360) data points in total.
+- For the last 1 day, keep 1 data point per 10 minutes,  (23 * 6 = 198) data points in total.
+- For the last 1 week, keep 1 data point per 1 hour, (6 * 24 = 144) data points in total.
+
+The total number of downsampling data points should not exceed 700.
```
[the remark this answers is not in the document]
Comment (zmstone) [reply active/0031-new-monitor-histogram.md]: updated algorithm

**Pull request closed without merging**